docs/_clef/Rules.md

```diff
@@ -48,98 +87,171 @@ function ApproveListing(req){
-an attack surface -- if an attacker can control the ruleset, a much simpler attack would be to implement an "always-approve" rule instead of exploiting the js vm. The only benefit
-to be gained from attacking the actual `signer` process from the `js` side would be if it could somehow extract cryptographic keys from memory.
+### The Javascript VM
+The downside of the very flexible rule implementation included in Clef is
+that the `signer` binary needs to contain a Javascript engine. This is an
+additional attack surface. The only viable attack is for an adversary to 
+somehow extract cryptographic keys from memory during the Javascript VM execution.
+The hash-based rule attestation condition means the actual Javascript code 
+executed by the Javascript engine is not a viable attack surface. A much simpler
+attack would be to surreptitiously insert an attested "always-approve" rule 
+instead of attempting to exploit the Javascript virtual machine. The Javascript
+engine is quite simple to implement and there are currently no known security
+vulnerabilities, not have there been any security problems identified for the
+similar Javascript VM implemented in Geth.
 
-##### Security in usability
+### Writing rules
 
-Javascript is flexible, but also easy to get wrong, especially when users assume that `js` can handle large integers natively. Typical errors
-include trying to multiply `gasCost` with `gas` without using `bigint`:s.
+Since the user has complete freedom to write custom rules, it is plausible that those rules
+could create unintended security vulnerabilities. This can only really be protected by 
+coding very carefully and trying to test rulesets (e.g. on a private testnet) before 
+implementing them on a public network.
 
-It's unclear whether any other DSL could be more secure; since there's always the possibility of erroneously implementing a rule.
+Javascript is very flexible but also easy to write incorrectly. For example, users
+might assume that javascript can handle large integers natively rather than explicitly
+using `bigInt`. This is an error commonly encountered in the Ethereum context when 
+users attempt to multiply `gas` by `gasCost`.
 
+It’s unclear whether any other language would be more secure - there is alwas the possibility
+of implementing an insecure rule.
 
-## Credential management
+### File security
 
-The ability to auto-approve transaction means that the signer needs to have necessary credentials to decrypt keyfiles. These passwords are hereafter called `ksp` (keystore pass).
+A ruleset should have its file permissions set to prevent attacks where an adversary
+drops files onto the user's disk. Specifically, `ruleset.js` should only be loaded if it
+is set to `readonly` (`r-xx-xx-x`). To update the ruleset, it should be explicitly made
+writeable then set back to `readonly` before being reloaded.
```

Comment:
Please delete this. The original doc included this as "potential todos", but then instead we went with the "attestation" solution. So the attack of replacing the file on disk is moot.

Reply:
```suggestion
```